src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -1719,29 +1718,28 @@ private void getDestinationMetadata() throws SQLServerException {
             destColumnCount = rs.getMetaData().getColumnCount();
             destColumnMetadata = new HashMap<>();
             destCekTable = rs.getCekTable();
-
-            if (!connection.getServerSupportsColumnEncryption()) {
-                // SQL server prior to 2016 does not support encryption_type
-                rsMoreMetaData = ((SQLServerStatement) connection.createStatement())
-                        .executeQueryInternal("select collation_name from sys.columns where " + "object_id=OBJECT_ID('"
-                                + Util.escapeSingleQuotes(destinationTableName) + "') " + "order by column_id ASC");
-            } else {
-                rsMoreMetaData = ((SQLServerStatement) connection.createStatement()).executeQueryInternal(
-                        "select collation_name, encryption_type from sys.columns where " + "object_id=OBJECT_ID('"
-                                + Util.escapeSingleQuotes(destinationTableName) + "') " + "order by column_id ASC");
-            }
-            for (int i = 1; i <= destColumnCount; ++i) {
-                if (rsMoreMetaData.next()) {
-                    if (!connection.getServerSupportsColumnEncryption()) {
-                        destColumnMetadata.put(i, new BulkColumnMetaData(rs.getColumn(i),
-                                rsMoreMetaData.getString("collation_name"), null));
+            try (SQLServerStatement statementMoreMetadata = (SQLServerStatement) connection.createStatement();
+                    SQLServerResultSet rsMoreMetaData = (!connection
+                            .getServerSupportsColumnEncryption() ? statementMoreMetadata
+                                    .executeQueryInternal("select collation_name from sys.columns where "
+                                            + "object_id=OBJECT_ID('" + Util.escapeSingleQuotes(destinationTableName) + "') "
```

Comment:
This is less readable to me. Maybe you could only "if else" the query string?

Reply:
Then we can't place it in try with resources block. Also will need another try block, and finally block to manually close it.
I don't see any issues - its a simple ternary operator.